src/platform_impl/web/event_loop/window_target.rs

```diff
@@ -163,5 +162,23 @@ impl<T> WindowTarget<T> {
                 },
             });
         });
+
+        let runner = self.runner.clone();
+        let event_canvas = canvas.clone();
```

Comment:
Could we avoid cloning the whole canvas if we create the `intended_size` here and move it into the closure? We could  keep there the size before entering fullscreen by reading the size of the `raw` canvas.

Reply:
We do need to keep track of `intended_size` as it changes, because it can be changed by the user at any point.

Reply:
I was thinking of ignoring user resizes when fullscreen is enabled. Does that break something? Or would it be unintuitive?

Reply:
That works for when fullscreen is enabled but the problem comes in when you have something like:

```
create window 1 at size of 800x600
set window 1's size to 640x480
fullscreen window 1
un-fullscreen window 1
```

You probably want it to return to `640x480`, which means the intended size can change during the lifetime of the program.

Reply:
Couldn't we clone `raw` instead and use it to read the `intended_size`, store it while fullscreen mode is enabled, and restore the correct size when disabled? We could wrap the raw canvas in a new type if necessary. It's just that implementing `Clone` for the current `Canvas` type feels a bit hacky because of all the event listeners.

Reply:
Great point, I used that design and was able to remove the Clone impl from `Canvas`.